components/server/src/user/user-controller.ts

```diff
@@ -267,9 +267,9 @@ export class UserController {
 
                 const token = instance.status.ownerToken;
                 if (!token) {
-                    // no token, no problem. The dashboard will try again later.
-                    res.sendStatus(200);
-                    log.debug("attempted to fetch workspace cookie, but instance has no owner token", {
+                    // There is no token to answer with, so we sent a 404. The client has to properly handle this case with retries/timeouts, etc.
+                    res.sendStatus(404);
```

Comment:
Why 404? If there's no token, it sounds like an authorization issue so I'd expect a 401?

Reply:
It's in line with [R243](https://github.com/gitpod-io/gitpod/pull/15610/files#diff-0fda9115576b7e1168858f5d4f8869e3120c3a8364cc8f43335037ebb78f6734R243) and fits into the context of `GET /auth/workspace-cookie`.